subsys/net/lib/websocket/websocket.c

```diff
@@ -443,12 +443,25 @@ static int websocket_ioctl_vmeth(void *obj, unsigned int request, va_list args)
 }
 
 static void websocket_mask_payload(u8_t *payload, size_t payload_len,
-				   u32_t masking_value)
+				   u32_t masking_value, s8_t mask_shift)
```

Comment:
Are you sure this argument should be of s8_t type, instead of int? Some compilers may produce less efficient code in thus case (effectively, adding sign extension cast from s8_t to full register size where all operations are handled then).